public_html/wp-content/plugins/camptix-attendee-survey/camptix-attendee-survey.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * Plugin Name:     CampTix Attendee Survey
+ * Plugin URI:      https://wordcamp.org
+ * Description:     Send survey to WordCamp attendees.
+ * Author:          WordCamp.org
+ * Author URI:      https://wordcamp.org
+ * Version:         1
+ *
+ * @package         CampTix\AttendeeSurvey
+ */
+
+namespace CampTix\AttendeeSurvey;
+
+use function CampTix\AttendeeSurvey\Email\{add_email, delete_email};
+use function CampTix\AttendeeSurvey\Page\{add_page, delete_page};
+
+defined( 'WPINC' ) || die();
+
+/**
+ * Local dependencies.
+ */
+require_once get_includes_path() . 'email.php';
+require_once get_includes_path() . 'page.php';
+
+/**
+ * Plugin activation and deactivation hooks.
+ */
+register_deactivation_hook( __FILE__, __NAMESPACE__ . '\deactivate' );
+
+/**
+ * Actions & hooks
+ */
+add_action( 'plugins_loaded', __NAMESPACE__ . '\load' );
+
+/**
+ * Get the ID of the survey feature.
+ */
+function get_feature_id() {
+	return 'attendee_survey';
+}
+
+/**
+ * Check dependencies for loading the plugin.
+ *
+ * @return bool
+ */
+function can_load() {
+	$skip_feature = wcorg_skip_feature( get_feature_id() );
+
+	return ! $skip_feature;
+}
+
+/**
+ * Include the rest of the plugin.
+ *
+ * @return void
+ */
+function load() {
+	if ( ! can_load() ) {
+		return;
+	}
+
+	// We only want to admin panel on central, nothing else.
+	if ( WORDCAMP_ROOT_BLOG_ID === get_current_blog_id() ) {
+		require_once get_includes_path() . 'admin-page.php';
+	}
+
+	if ( ! is_wordcamp_type( 'next-gen' ) ) {
+		return;
+	}
+
+	require_once get_includes_path() . 'cron.php';
+
+	add_action( 'init', __NAMESPACE__ . '\maybe_activate_on_current_site' );
+}
+
+/**
+ * The activation routine for a single site.
+ *
+ * @return void
+ */
+function maybe_activate_on_current_site() {
+
+	// Currently only running this for events that are next-gen.
+	if ( ! is_wordcamp_type( 'next-gen' ) ) {
+		return;
+	}
+
+	add_page();
+	add_email();
+
+	// Flushing the rewrite rules is buggy in the context of `switch_to_blog`.
+	// The rules will automatically get recreated on the next request to the site.
+	delete_option( 'rewrite_rules' );
+}
```

Comment:
What do you think of just keeping one `if ( ! is_wordcamp_type( 'next-gen' ) ) {` condition? Otherwise, wouldn't maintenance require changes in both places in the future? From the name `maybe_activate_on_current_site()`, I can understand your original thought, but it seems to me this way is clearer.

```suggestion
function load() {
	if ( ! can_load() ) {
		return;
	}

	// We only want the admin panel on central, nothing else.
	if ( WORDCAMP_ROOT_BLOG_ID === get_current_blog_id() ) {
		require_once get_includes_path() . 'admin-page.php';
	}

	if ( is_wordcamp_type( 'next-gen' ) ) {
		require_once get_includes_path() . 'cron.php';
		add_action( 'init', __NAMESPACE__ . '\maybe_activate_on_current_site' );
	}
}

/**
 * The activation routine for a single site.
 *
 * @return void
 */
function maybe_activate_on_current_site() {
	add_page();
	add_email();

	// Flushing the rewrite rules is buggy in the context of `switch_to_blog`.
	// The rules will automatically get recreated on the next request to the site.
	delete_option( 'rewrite_rules' );
}
```

Reply:
Ah right. Yeah, one in `maybe_activate_on_current_site` was because it was also triggered by an activation hook removed in [b444446](https://github.com/WordPress/wordcamp.org/pull/997/commits/b444446a0d2dc79c5348fc1e5a439c367bb14867).